src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.SpeculativeMemberSemanticModel.cs

```diff
@@ -22,14 +23,14 @@ public SpeculativeMemberSemanticModel(SyntaxTreeSemanticModel parentSemanticMode
             {
             }
 
-            protected override BoundNode RewriteNullableBoundNodes(BoundNode boundRoot, Binder binder, DiagnosticBag diagnostics)
+            protected override BoundNode RewriteNullableBoundNodesWithCheckpoints(BoundNode boundRoot, Binder binder, DiagnosticBag diagnostics, Dictionary<BoundNode, NullableWalker.Checkpoint> checkpointMap)
             {
                 // https://github.com/dotnet/roslyn/issues/35037: Speculative models are going to have to do something more advanced
                 // here. They will need to run nullable analysis up to the point that is being speculated on, and
                 // then take that state and run analysis on the statement or expression being speculated on.
                 // Currently, it will return incorrect info because it's just running analysis on the speculated
                 // part.
-                return NullableWalker.AnalyzeAndRewrite(Compilation, MemberSymbol as MethodSymbol, boundRoot, binder, diagnostics);
+                return NullableWalker.AnalyzeAndRewrite(Compilation, MemberSymbol as MethodSymbol, boundRoot, binder, diagnostics, checkpointMapOpt: null);
```

Comment:
In a follow-up PR I will be asserting that a checkpoint map isn't passed to this version of the semantic model.